lib/src/implicitly_animated_reorderable_list.dart

```diff
@@ -491,20 +445,49 @@ class ImplicitlyAnimatedReorderableListState<E>
     });
   }
 
-  void onDragEnded() {
-    if (dragKey == null || _closestList.isEmpty) return;
+  ///
+  /// Finding destination of drag item.
+  /// Items can move to up/down so position can be changed while index is not change.
+  /// We need virtual index reflecting moves.
+  /// Empty space of virtual indexes for non-drag item will be destination index of drag item.
+  ///
+
+  _Item findSwapTargetItem() {
+    final currentIndexList = _itemBoxes.values.where((item) => item != dragItem).map((item) {
+      final translation = getTranslation(item.key);
 
-    if (getTranslation(_next.key) == 0.0) {
-      _dispatchMove(_next.key, _up ? _dragSize : -_dragSize);
+      if (translation  == 0) {
+        return item.index;
+      } else if (translation > 0) {
+        return item.index + 1;
+      } else {
+        return item.index - 1;
+      }
+    });
+
+    int dragTargetIndex = _dragIndex;
+    for (int i = 0; i < _itemBoxes.length; i++) {
+      if (!currentIndexList.contains(i)) {
+        dragTargetIndex = i;
+        break;
+      }
     }
 
+    return _itemBoxes.values.firstWhere((item) => item.index == dragTargetIndex);
```

Comment:
```suggestion
    return _itemBoxes.values.firstWhere(
      (item) => item.index == dragTargetIndex,
      orElse: () => dragItem,
    );
```
`firstWhere` throws `StateError` when it couldn't find an item. So just to be sure.

Reply:
thanks!